greenpithumb/db_store.py

```diff
@@ -41,17 +41,20 @@ def latest_soil_moisture(self):
             return results[0][0]
 
     def retrieve_soil_moisture(self):
-        """Returns timestamp and soil moisture readings.
+        """Retrieves timestamp and soil moisture readings.
 
-        Return values are a list of namedtuples with 'timestamp' and
-        'soil_moisture' fields.
+        Returns:
+            A list of objects with 'timestamp' and 'soil_moisture' fields.
         """
-        self._cursor.execute("SELECT * FROM soil_moisture")
+        self._cursor.execute('SELECT * FROM soil_moisture')
         SoilMoistureRecord = collections.namedtuple(
             'SoilMoistureRecord', ['timestamp', 'soil_moisture'])
         soil_moisture_data = map(SoilMoistureRecord._make,
                                  self._cursor.fetchall())
-        return soil_moisture_data
+        if soil_moisture_data == []:
+            return None
```

Comment:
Suggest returning `[]` here since the library is easier to use if we guarantee the result is always a list (like caller can always iterate it, always check the length).